src/App.js

```diff
@@ -1,23 +1,62 @@
+import React from 'react';
 import './App.css';
+import axios from 'axios';
+import CityDisplay from './CityDisplay';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      searchQuery: '',
+      location: {},
+      lat: '',
+      lon: ''
+    }
+  }
+
+  handleChange = (event) => {
+    this.setState({ searchQuery: event.target.value })
+  }
+
+  getLocation = async () => {
+    const url = `https://us1.locationiq.com/v1/search.php?key=${process.env.REACT_APP_LOCATIONIQ_KEY}&q=${this.state.searchQuery}&format=json`;
+    console.log('URL:', url);
+    //call axios
+    const response = await axios.get(url);
+    console.log('response object: ', response);
+    console.log('response.data[0]: ', response.data[0]);
+    this.setState({ location: response.data[0] });
+    this.setState({ lat: response.data[0].lat });
+    this.setState({ lon: response.data[0].lon });
```

Comment:
When the lat and lon values can be read from the location {} state variable why set additional state variables to hold these two data values? Might be able to remove them to eliminate small but redundant memory usage.

Reply:
Great idea, Manuch! I'll review this later today.